### PR TITLE
Ensure the correct site goals option is returned

### PR DESCRIPTION
**Pull request by heavyweight**
In our site base we had a getter for the `siteGoals` option which is not used or returned anywhere. This PR modifies this to be using the correct option which is `site_goals` and wires it to be returned with the site response.

Fixes #

## Proposed changes:

* Wire `site_goals` to be returned for a site

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Sync this PR to your sandbox
* Follow the onboarding flow at WordPress.com/start
* Once you arrive at the `goals` step select a few there
* Once you submit them check on your sandbox if they are properly set 
`wp option get site_goals --blog=BLOG_ID`
* Reload the page and inspect the site request. The `site_goals` should be a part of the response.

### CODE REVIEW

File: projects/plugins/jetpack/sal/class.json-api-site-base.php [hunk 1]
```diff
@@ -1348,15 +1348,23 @@ public function get_design_type() {
...
+		if ( ! empty( $options['siteGoals'] ) ) {
+			return $options['siteGoals'];
+		}
+
+		if ( ! empty( $options['site_goals'] ) && is_array( $options['site_goals'] ) ) {
+			return $options['site_goals'];
+		}
+
+		return null;
+	}
```
Comment (alshakero) [inline projects/plugins/jetpack/sal/class.json-api-site-base.php]: This is a breaking change. If somebody is consuming this function before your change, they'd expect a string or null, never an array. Now they will get an array and something may throw.

I did search though and nothing across Automattic is using this function at all. So maybe we should nuke `siteGoals` and keep only `site_goals: array|null`?

**Review by alshakero — DISMISSED** [review]
Left one note.

File: projects/plugins/jetpack/sal/class.json-api-site-base.php (hunk 1 above)
Comment (escapemanuele) [reply projects/plugins/jetpack/sal/class.json-api-site-base.php]: Yeah it was mostly my paranoia about nuking something that I'm not 100% sure is not being used, being this Jetpack.


If we say that it isn't a problem, then let's go with nuking things!

Comment (heavyweight) [reply projects/plugins/jetpack/sal/class.json-api-site-base.php]: I was also trying to find where it is used but didn't. Looks like leftover code from way back .
I prefer to remove `siteGoals` and return empty array or array of goals.

Comment (escapemanuele) [reply projects/plugins/jetpack/sal/class.json-api-site-base.php]: Ok, cool 👌

**Review by escapemanuele — APPROVED** [review]
Thanks, we'll make good use of `site_goals`!

**Pull request merged**